### ISSUE TITLE
Map attribution not clickable

### ISSUE THREAD

**Issue by kylebarron** [visgl/deck.gl#4165]
#### Description
Map attribution not clickable. 

The Mapbox [attribution page](https://docs.mapbox.com/help/how-mapbox-works/attribution/#text-attribution) says 

> You must properly attribute **and link** Mapbox and OpenStreetMap when using the Mapbox Streets tileset

(emphasis mine).

I'm not affiliated with Mapbox, but it still seems like it would be nice to have each attribution source be a hyperlink.

#### Repro Steps
In any of the published examples, for example [`LineLayer`](https://deck.gl/#/examples/core-layers/line-layer), I can't click on the attribution link. For example, `(c) Mapbox` should be clickable, but when I try to click, it moves the map instead.
![image](https://user-images.githubusercontent.com/15164633/72642528-9a6de080-393a-11ea-8eb4-ffa45e38852b.png)

**Comment by Pessimistress** [comment]
Because deck.gl renders a transparent canvas on top of mapbox, it is not possible to click on the hyperlink in mapbox's container. There is not much we can do. Maybe Mapbox can offer a standalone attribution control.

**Comment by kylebarron** [comment]
Ok, that seems to make sense. Would it be best to create an issue on `mapbox-gl-js` then?

**Comment by Pessimistress** [comment]
Yes, I think we need Mapbox's help to fix this. There are two possible solutions:
- Instead of hyperlinks, use the `click` listener. That way if we disable `pointer-events` on the canvas, clicking on the links will still open the attribution page.
- Allow its attribution control to be rendered into a user-provided container. Then you can position a div above the canvas for the links to go in.

**Comment by kylebarron** [comment]
Ok thanks, I created the linked issue in mapbox/mapbox-gl-js

**Comment by andrewharvey** [comment]
Is this a mapbox-gl-js issue?

See here in the linked example using deck.gl, the div which contains the whole Mapbox map div (including canvas and controls) is `z-index: -1`, and then the deck.gl canvas on top is `z-index: 1`, which is what's causing the issue.

I would have thought the correct solution would be to insert the deck.gl canvas in between the Mapbox canvas and the Mapbox controls (either by using z-index or by inserting the deck.gl canvas in between in the DOM).

### with deck.gl
![Screenshot from 2020-01-18 22-28-36](https://user-images.githubusercontent.com/117278/72663019-ea839a80-3a41-11ea-98af-88217faa9686.png)

### stock mapbox gl js
![Screenshot from 2020-01-18 22-25-43](https://user-images.githubusercontent.com/117278/72663020-eb1c3100-3a41-11ea-917b-432de5d12a06.png)

**Comment by Pessimistress** [comment]
> Is this a mapbox-gl-js issue?

It is not a mapbox-gl-js “issue” per se, more like a feature request. I’m afraid we can’t use the stock mapbox gl js solution. To name a few use cases, We need to support using one deck canvas with two synced based maps side by side; or a main 3D scene with a mini-map; or a “free” camera that allows pitch angle > 60, in which case we need to hide the base map but keep rendering the 3D scene.

deck.gl can also be used with other base map solutions, and the more hard-coded special treatment we do the more likely it’s going to break.

**Comment by scottpietka** [comment]
@Pessimistress et al, thank you for working on this issue. I'm going to open an Engineering Support Ticket and work with my team to see what solutions exist. Stay tuned for an update from our team.

**Comment by sgolbabaei** [comment]
Hello, 
I am investigating potential fixes to this issue in mapbox-gl-js. Is this problem happening solely in the example page?
I cloned the deck.gl version 7.0-release and I didn't see this problem (the glass element preventing the user to click on the attribution) in my local server.
Are there any other releases that are being effected? If so, it would be very helpful if there was a minimal example demonstrating the issue.

**Comment by Pessimistress** [comment]
@sgolbabaei the latest release branch is 8.0-release. 7.0 may be using an old version of mapbox.

**Comment by sgolbabaei** [comment]
@Pessimistress I tried the 8.0-release version locally and still I am able to click on the attribution control on my local server..
![8 0release](https://user-images.githubusercontent.com/32684731/75581437-503e4b80-5a1e-11ea-8585-48887b235ea0.gif)

Is this problem happening solely in the example page? it would be very helpful if there was a minimal example demonstrating the issue.

**Comment by kylebarron** [comment]
It doesn't work for me with the deck examples on master
![Screen Recording 2020-02-28 at 2 23 26 PM](https://user-images.githubusercontent.com/15164633/75588558-2b55d280-5a36-11ea-8ff9-034ab55da547.gif)

```bash
git clone https://github.com/uber/deck.gl
cd deck.gl
npm run bootstrap
cd examples/website/geojson
npm install
export MapboxAccessToken=...
npm run start-local
```

**Comment by sgolbabaei** [comment]
thanks @kylebarron for providing the example.

**Comment by shawnmmatthews** [comment]
I can confirm I'm seeing the same results with an IconLayer in 8.1. Both the attribution and navigation control are losing a z-index fight to the DeckGL overlay. The offending code seems to be the z-index: -1 nested inside of div#view-default-view.

**Comment by kylebarron** [comment]
You can fix the navigation control by putting it in its own `div`, and passing `ContextProvider={MapContext.Provider}` to `Deck.gl`.
:
```jsx
import {
  _MapContext as MapContext,
  NavigationControl
} from 'react-map-gl';
...
        <DeckGL
          ContextProvider={MapContext.Provider}
		>

          <div style={{ position: 'absolute', right: 30, top: 110, zIndex: 1 }}>
            <NavigationControl />
          </div>
        </DeckGL>
```

I think the point of this discussion here is that we'd like an `AttributionControl` element like `NavigationControl`.

**Comment by kylebarron** [comment]
Just reading through this again...

> Maybe Mapbox can offer a standalone attribution control.

@Pessimistress React Map GL already offers a `ScaleControl`, `NavigationControl`, `GeolocateControl`, etc. The Mapbox GL JS API additionally includes an [`AttributionControl`](https://docs.mapbox.com/mapbox-gl-js/api/#attributioncontrol). How hard would it be to wrap the Mapbox `AttributionControl` into a React component in React Map GL?

Then it would seem possible to do:
```jsx
import {
  _MapContext as MapContext,
  AttributionControl
} from 'react-map-gl';
...
        <DeckGL
          ContextProvider={MapContext.Provider}
		>

          <div style={{ position: 'absolute', bottom: 0, right: 0, zIndex: 1 }}>
            <AttributionControl />
          </div>
        </DeckGL>
```

It should then be clickable in the same way that the `NavigationControl` currently is. Does that sound feasible?

**Comment by Pessimistress** [comment]
react-map-gl is adding a new `AttributionControl` in v5.3/v6.1. It is currently available in `6.1.0-beta.1`.

See usage example with DeckGL here: https://github.com/visgl/react-map-gl/blob/master/examples/deckgl-overlay/src/app.js
